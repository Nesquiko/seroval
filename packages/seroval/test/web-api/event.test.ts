import { describe, it, expect } from 'vitest';
import {
  crossSerialize,
  crossSerializeAsync,
  crossSerializeStream,
  deserialize,
  fromJSON,
  serialize,
  serializeAsync,
  toJSON,
  toJSONAsync,
} from '../../src';

const EXAMPLE_EVENT_TYPE = 'example';

describe('Event', () => {
  describe('serialize', () => {
    it('supports Event', () => {
      const example = new Event(EXAMPLE_EVENT_TYPE);
      const result = serialize(example);
      expect(result).toMatchSnapshot();
      const back = deserialize<Event>(result);
      expect(back).toBeInstanceOf(Event);
    });
  });
  describe('serializeAsync', () => {
    it('supports Event', async () => {
      const example = new Event(EXAMPLE_EVENT_TYPE);
      const result = await serializeAsync(Promise.resolve(example));
      expect(result).toMatchSnapshot();
      const back = await deserialize<Promise<Event>>(result);
      expect(back).toBeInstanceOf(Event);
    });
  });
  describe('toJSON', () => {
    it('supports Event', () => {
      const example = new Event(EXAMPLE_EVENT_TYPE);
      const result = toJSON(example);
      expect(JSON.stringify(result)).toMatchSnapshot();
      const back = fromJSON<Event>(result);
      expect(back).toBeInstanceOf(Event);
    });
  });
  describe('toJSONAsync', () => {
    it('supports Event', async () => {
      const example = new Event(EXAMPLE_EVENT_TYPE);
      const result = await toJSONAsync(Promise.resolve(example));
      expect(JSON.stringify(result)).toMatchSnapshot();
      const back = await fromJSON<Promise<Event>>(result);
      expect(back).toBeInstanceOf(Event);
    });
  });
  describe('crossSerialize', () => {
    it('supports Event', () => {
      const example = new Event(EXAMPLE_EVENT_TYPE);
      const result = crossSerialize(example);
      expect(result).toMatchSnapshot();
    });
    describe('scoped', () => {
      it('supports Event', () => {
        const example = new Event(EXAMPLE_EVENT_TYPE);
        const result = crossSerialize(example, { scopeId: 'example' });
        expect(result).toMatchSnapshot();
      });
    });
  });
  describe('crossSerializeAsync', () => {
    it('supports Event', async () => {
      const example = new Event(EXAMPLE_EVENT_TYPE);
      const result = await crossSerializeAsync(Promise.resolve(example));
      expect(result).toMatchSnapshot();
    });
    describe('scoped', () => {
      it('supports Event', async () => {
        const example = new Event(EXAMPLE_EVENT_TYPE);
        const result = await crossSerializeAsync(Promise.resolve(example), { scopeId: 'example' });
        expect(result).toMatchSnapshot();
      });
    });
  });
  describe('crossSerializeStream', () => {
    it('supports Event', async () => new Promise<void>((done) => {
      const example = new Event(EXAMPLE_EVENT_TYPE);
      crossSerializeStream(Promise.resolve(example), {
        onSerialize(data) {
          expect(data).toMatchSnapshot();
        },
        onDone() {
          done();
        },
      });
    }));
    describe('scoped', () => {
      it('supports Event', async () => new Promise<void>((done) => {
        const example = new Event(EXAMPLE_EVENT_TYPE);
        crossSerializeStream(Promise.resolve(example), {
          scopeId: 'example',
          onSerialize(data) {
            expect(data).toMatchSnapshot();
          },
          onDone() {
            done();
          },
        });
      }));
    });
  });
});
