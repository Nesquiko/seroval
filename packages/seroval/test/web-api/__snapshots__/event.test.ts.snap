// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Event > crossSerialize > scoped > supports Event 1`] = `"($R=>$R[0]=new Event(\\"example\\",$R[1]={bubbles:!1,cancelable:!1,composed:!1}))($R[\\"example\\"])"`;

exports[`Event > crossSerialize > supports Event 1`] = `"$R[0]=new Event(\\"example\\",$R[1]={bubbles:!1,cancelable:!1,composed:!1})"`;

exports[`Event > crossSerializeAsync > scoped > supports Event 1`] = `"($R=>$R[0]=Promise.resolve($R[1]=new Event(\\"example\\",$R[2]={bubbles:!1,cancelable:!1,composed:!1})))($R[\\"example\\"])"`;

exports[`Event > crossSerializeAsync > supports Event 1`] = `"$R[0]=Promise.resolve($R[1]=new Event(\\"example\\",$R[2]={bubbles:!1,cancelable:!1,composed:!1}))"`;

exports[`Event > crossSerializeStream > scoped > supports Event 1`] = `"($R=>$R[0]=_$.P())($R[\\"example\\"])"`;

exports[`Event > crossSerializeStream > scoped > supports Event 2`] = `"($R=>_$.Ps($R[0],$R[1]=new Event(\\"example\\",$R[2]={bubbles:!1,cancelable:!1,composed:!1})))($R[\\"example\\"])"`;

exports[`Event > crossSerializeStream > supports Event 1`] = `"$R[0]=_$.P()"`;

exports[`Event > crossSerializeStream > supports Event 2`] = `"_$.Ps($R[0],$R[1]=new Event(\\"example\\",$R[2]={bubbles:!1,cancelable:!1,composed:!1}))"`;

exports[`Event > serialize > supports Event 1`] = `"new Event(\\"example\\",{bubbles:!1,cancelable:!1,composed:!1})"`;

exports[`Event > serializeAsync > supports Event 1`] = `"Promise.resolve(new Event(\\"example\\",{bubbles:!1,cancelable:!1,composed:!1}))"`;

exports[`Event > toJSON > supports Event 1`] = `"{\\"t\\":{\\"t\\":37,\\"i\\":0,\\"s\\":\\"example\\",\\"f\\":{\\"t\\":10,\\"i\\":1,\\"p\\":{\\"k\\":[\\"bubbles\\",\\"cancelable\\",\\"composed\\"],\\"v\\":[{\\"t\\":2,\\"s\\":3},{\\"t\\":2,\\"s\\":3},{\\"t\\":2,\\"s\\":3}],\\"s\\":3},\\"o\\":0}},\\"f\\":16383,\\"m\\":[]}"`;

exports[`Event > toJSONAsync > supports Event 1`] = `"{\\"t\\":{\\"t\\":12,\\"i\\":0,\\"s\\":1,\\"f\\":{\\"t\\":37,\\"i\\":1,\\"s\\":\\"example\\",\\"f\\":{\\"t\\":10,\\"i\\":2,\\"p\\":{\\"k\\":[\\"bubbles\\",\\"cancelable\\",\\"composed\\"],\\"v\\":[{\\"t\\":2,\\"s\\":3},{\\"t\\":2,\\"s\\":3},{\\"t\\":2,\\"s\\":3}],\\"s\\":3},\\"o\\":0}}},\\"f\\":16383,\\"m\\":[]}"`;
